content/5_Plat/Convex_Hull.mdx

```diff
@@ -85,22 +84,205 @@ export const problems = {
 
 <!-- [convex hull](https://en.wikipedia.org/wiki/Convex_hull_algorithms). -->
 
+
+## Introduction
+
+The **Convex Hull** is the subset of points that forms the smallest convex polygon which encloses all points in the set.
+To visualize this, imagine that each point is a pole. Then, imagine what happens if you were to wrap a rope around the outside of all the poles, and then pull infinitely hard, such that the connections between any two points that lie on the edge of the rope are lines. The set of points that touch the rope (and are not colinear) is the convex hull.
+
+<Spoiler title="Convex Hull Visualization">
+
+![Convex Hull Visualization](./assets/convexhull.jpg)
+
+</Spoiler>
+
 <FocusProblem problem={problems.sample[0]} />
 
 ## With Graham Scan
 
-- [cp-algo](https://cp-algorithms.com/geometry/grahams-scan-convex-hull.html)
-- [My Implementation](<https://github.com/bqi343/USACO/blob/master/Implementations/content/geometry%20(13)/Polygons/ConvexHull2.h>)
+<Resources>
+	<Resource source="cp-algo" title="Graham Scan" url="https://cp-algorithms.com/geometry/grahams-scan-convex-hull.html" starred />
+	<Resource source="Benq" title="Graham Scan Implementation" url="https://github.com/bqi343/USACO/blob/master/Implementations/content/geometry%20(13)/Polygons/ConvexHull2.h" />
```

Comment:
Link to this implementation in the solution section instead.

(In general, I think implementations should be kept separate from other resources.)

Reply:
ok i moved the whole section to the soluton lmk wyt

Reply:
think it looks better to have the single link in a resources table by itself